addons/web/tests/test_db_manager.py

```diff
@@ -110,10 +117,72 @@ def test_database_duplicate(self):
         self.assertDbs([test_db_name])
 
         # delete the created database
-        res = self.session.post(self.url('/web/database/drop'), data={
-            'master_pwd': self.password,
-            'name': test_db_name,
-        }, allow_redirects=False)
-        self.assertEqual(res.status_code, 303)
+        res = self.url_open_drop(test_db_name)
         self.assertIn('/web/database/manager', res.headers['Location'])
         self.assertDbs([])
+
+    @freezegun.freeze_time('2023-12-26 15:30:00')
+    def test_database_restore(self):
+        test_db_name = self.db_name + '-test-database-restore'
+        self.assertNotIn(test_db_name, self.list_dbs_filtered())
+
+        # backup the current database inside a temporary zip file
+        res = self.session.post(
+            self.url('/web/database/backup'),
+            data={
+                'master_pwd': self.password,
+                'name': self.db_name,
+            },
+            allow_redirects=False,
+            stream=True,
+        )
+        res.raise_for_status()
+        self.assertEqual(
+            res.headers.get('Content-Disposition'),
+            f"attachment; filename*=UTF-8''{self.db_name}_2023-12-26_15-30-00.zip"
```

Comment:
I would use the cr.now to name the file and make an assertion on that. 
This will remove the need to use freezegun.

Reply:
The backup route is `auth="none"` meaning that there can be no cursor. Using `cr.now()` inside the controller is not possible and `datetime.datetime.utcnow().strftime("%Y-%m-%d_%H-%M-%S")` is used instead. I've used an assertRegex instead to void the need for freezegun.